api/src/main/java/org/apache/iceberg/Schema.java

```diff
@@ -381,6 +381,22 @@ public String idToAlias(Integer fieldId) {
     return null;
   }
 
+  /**
+   * Returns the index of the given field id.
+   *
+   * @param fieldId a column id in this schema
+   * @return the index of the field in the schema, or -1 if one wasn't found
+   */
+  public int idToIndex(Integer fieldId) {
```

Comment:
`idToPosition`?  `position` might be more clear. 

Also we are using `Integer` arg type, which means it can be null. Probably following the pattern from the `idToAlias ` above. But I am wondering if the `fieldId` arg should be primitive `int` type. 

Also, should the return type be `Integer` and nullable just to be consistent with other methods in this class?

Reply:
Thanks for the name suggestion. There are always performance concerns about non-primitive type. I'm kind of OK with both primitive type and non-primitive type here. In terms of consistency, these two public methods also return `int`. I will leave the return type to `int` unless we have a strong reason to use Integer.
```
public int schemaId()
public int highestFieldId()
```

Reply:
Even this place uses `int`, I'd prefer to use `int` for the parameter as well. It is unnecessary to do this boxing and unboxing.
https://github.com/apache/iceberg/blob/44c1d00b37f0d7fc2e978baad4f7a861a8335cf0/api/src/main/java/org/apache/iceberg/types/Types.java#L479

Reply:
> Even this place uses `int`, I'd prefer to use `int` for the parameter as well. It is unnecessary to do this boxing and unboxing.
> 
> https://github.com/apache/iceberg/blob/44c1d00b37f0d7fc2e978baad4f7a861a8335cf0/api/src/main/java/org/apache/iceberg/types/Types.java#L479

That is different. For NestedField, fieldId will never be null. hence it returns primitive int, which makes sense. But here, we may not find the field with the provided id, we are returning a special value `-1`. If we follow the style of other find methods in the `Schema` class, we can see `aliasToId` returns null if no matching field is found.

Reply:
Thanks for the comment. With @szehon-ho's suggestion https://github.com/apache/iceberg/pull/4683#discussion_r869717071, I removed the the new method.